lib/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -635,6 +635,10 @@ static void printReturnOp(OpAsmPrinter *p, ReturnOp &op) {
   *p << ' ' << *op.getOperand(0) << " : " << op.getOperand(0)->getType();
 }
 
+static void printUnreachableOp(OpAsmPrinter *p, UnreachableOp &op) {
+  *p << op.getOperationName() << " : () -> ()";
```

Comment:
Don't print a trailing empty functional type, it isn't necessary.

Reply:
On that note, can we just inline this function into the op definition? It is small enough.